src/pybind/mgr/volumes/fs/operations/op_sm.py

```diff
@@ -1,68 +1,132 @@
 import errno
 
+from enum import Enum, unique
 from typing import Dict
 
+from .versions.subvolume_base import SubvolumeTypes
 from ..exception import OpSmException
 
-class OpSm(object):
-    INIT_STATE_KEY = 'init'
+@unique
+class SubvolumeStates(Enum):
+    STATE_INIT          = 'init'
+    STATE_PENDING       = 'pending'
+    STATE_INPROGRESS    = 'in-progress'
+    STATE_FAILED        = 'failed'
+    STATE_COMPLETE      = 'complete'
+    STATE_CANCELED      = 'canceled'
 
-    FAILED_STATE = 'failed'
-    FINAL_STATE  = 'complete'
-    CANCEL_STATE = 'canceled'
+    @staticmethod
+    def from_value(value):
+        if value == "init":
+            return SubvolumeStates.STATE_INIT
+        if value == "pending":
+            return SubvolumeStates.STATE_PENDING
+        if value == "in-progress":
+            return SubvolumeStates.STATE_INPROGRESS
+        if value == "failed":
+            return SubvolumeStates.STATE_FAILED
+        if value == "complete":
+            return SubvolumeStates.STATE_COMPLETE
+        if value == "canceled":
+            return SubvolumeStates.STATE_CANCELED
+
+        raise OpSmException(-errno.EINVAL, "invalid state '{0}'".format(value))
+
+@unique
+class SubvolumeActions(Enum):
```

Comment:
These are used outside the subvolume set of classes by the async_cloner. Hence it would be better to retain the prefix so as to be explicit in async_cloner.